prow/kube/config.go

```diff
@@ -92,16 +83,96 @@ func LoadClusterConfigs(kubeconfig, projectedTokenFile string) (map[string]rest.
 	} else {
 		localCfg.UserAgent = version.UserAgent()
 	}
-	if localCfg != nil && projectedTokenFile != "" {
+	if localCfg != nil && kc.ProjectedTokenFile != "" {
 		localCfg.BearerToken = ""
-		localCfg.BearerTokenFile = projectedTokenFile
-		logrus.WithField("tokenfile", projectedTokenFile).Info("Using projected token file")
+		localCfg.BearerTokenFile = kc.ProjectedTokenFile
+		logrus.WithField("tokenfile", kc.ProjectedTokenFile).Info("Using projected token file")
 	}
 
-	kubeCfgs, currentContext, err := kubeConfigs(kubeconfig)
-	if err != nil {
-		return nil, fmt.Errorf("kubecfg: %v", err)
+	var candidates []string
+	if kc.File != "" {
+		candidates = append(candidates, kc.File)
+	}
+	if kc.Dir != "" {
+		files, err := ioutil.ReadDir(kc.Dir)
+		if err != nil {
+			return nil, fmt.Errorf("kubecfg dir: %v", err)
+		}
+		var re *regexp.Regexp
+		for _, file := range files {
+			filename := file.Name()
+			if re == nil {
+				re, err = regexp.Compile(kc.RegExRaw)
```

Comment:
I still don't understand why we need this. Why don't users just give us a dir that *only* contains configs?

Reply:
> I still don't understand why we need this.

Replied https://github.com/kubernetes/test-infra/pull/23194#discussion_r687199595

> Why don't users just give us a dir that only contains configs?

We could do it. That would require us to split our big secret `build-farm-credentials` into small ones.
Currently, we select keys e.g.,

https://github.com/openshift/release/blob/f272c7cad1fd78a00cfc1f88bba250746e36fe5d/clusters/app.ci/prow/03_deployment/sinker.yaml#L142-L155

However, we do not want to change the deployment after adding the kubeconfig for the new cluster, right?
This flag is optional. Have you seen some troubles coming out of it alreaddy?

Or, Is splitting in your plan anyway?

Reply:
It feels silly to give the tool more KUBECONFIG entries than we need to, then ask the tool to filter them. We should provide only the ones that are necessary and load all of them.

Reply:
So, splitting the larger secret into a small one that has only the SA ones we need (for all farms) is correct here iMO

Reply:
I will remove the code for filtering.

The impact is described there:
https://issues.redhat.com/browse/DPTP-2412?focusedCommentId=18833694&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-18833694

Reply:
Updated.